### PR TITLE
Feat/add posts

### PR DESCRIPTION
**Pull request by Maianki**
### added posts CRUD And bookmarks
This PR includes:

- [ ] CRUD operation in posts
- [ ] Bookmarks route

### CODE REVIEW

File: src/components/Sidebar/SidebarLeft.jsx [hunk 1]
```diff
@@ -15,38 +15,83 @@ import {
...
+              <AiOutlineHome />
+              <Box px={2}>Home</Box>
+            </Flex>
+          </NavLink>
+          <NavLink
+            to='/explore'
+            className={({ isActive }) =>
+              isActive
+                ? `${styles.isNavLinkActive} ${styles.sidebarMenuNavLink}`
+                : `${styles.isNavLinkInactive} ${styles.sidebarMenuNavLink}`
```
Comment (anik31) [inline src/components/Sidebar/SidebarLeft.jsx:58]: suggestion - you can store the nav items in an array

File: src/services/postServices.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import axios from "axios";
+
+/**
+ * Service to return all posts
+ */
+export const getPostsService = async () => {
+  const response = await axios.get(POSTS_API);
+  return response;
+};
+
```
Comment (anik31) [inline src/services/postServices.js:11]: appreciation - liked that you've kept the services in a separate file

**Review by anik31 — COMMENTED** [review]
The code is good & well structured. But fix the issue in the view. Data is not getting displayed.

File: src/features/Post/Post.jsx [hunk 3]
```diff
@@ -1,28 +1,107 @@
...
+}) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { user: currentUser } = useSelector((state) => state.auth);
+  const { userToken: token } = useSelector((state) => state.auth);
+  const { allBookmarks } = useSelector((state) => state.users);
+
+  const dispatch = useDispatch();
+  const handleBookmarks = () => {
+    if (allBookmarks.some((post) => post._id === postId)) {
+      console.log("removed");
```
Comment (sainath7878) [inline src/features/Post/Post.jsx:42]: remove console.log

File: src/features/Post/postSlice.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  }
+);
+
+export const createUserPost = createAsyncThunk(
+  "posts/createUserPost",
+  async ({ content }, { rejectWithValue }) => {
+    const token = JSON.parse(localStorage.getItem("SGtoken"));
+    try {
+      const response = await createPostService(token, content);
+      console.log(response.data.posts);
```
Comment (sainath7878) [inline src/features/Post/postSlice.js]: remove console.log

File: src/features/Post/postSlice.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    }
+  }
+);
+
+export const deleteUserPost = createAsyncThunk(
+  "posts/deleteUserPost",
+  async ({ token, postId }, { rejectWithValue }) => {
+    try {
+      const response = await deletePostService(token, postId);
+      console.log(response);
```
Comment (sainath7878) [inline src/features/Post/postSlice.js:77]: remove console.log

File: src/services/usersService.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  );
+  return response;
+};
+
+/**
+ * Service to remove post from bookmarks
+ */
+export const removeFromBookmarksService = async (token, postId) => {
+  console.log("remove bookmarks", token, postId);
```
Comment (sainath7878) [inline src/services/usersService.js]: remove console

**Review by sainath7878 — APPROVED** [review]
Code looks clean and understandable. Check the functionality. Apart from the functionality everything looks good.

**Comment by Maianki** [comment]
Thanks @anik31 @sainath7878 . I have done the said changes.

**Pull request merged**